libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/StreamingExtensions/BotFrameworkStreamingExtensionsAdapter.cs

```diff
@@ -0,0 +1,197 @@
+        /// <param name="middleware">The middleware to add to the pipeline.</param>
+        /// <returns>This instance of the adapter.</returns>
+        public new BotFrameworkStreamingExtensionsAdapter Use(IMiddleware middleware)
+        {
+            MiddlewareSet.Use(middleware);
+            return this;
+        }
+
+        /// <summary>
+        /// Overload for processing activities when given an authheader.
+        /// </summary>
+        /// <param name="authHeader">The auth token provided by the request.</param>
+        /// <param name="activity">The activity to process.</param>
+        /// <param name="callback">The BotCallBackHandler to call on completion.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>The response to the activity.</returns>
+        public async Task<InvokeResponse> ProcessActivityAsync(string authHeader, Activity activity, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            BotAssert.ActivityNotNull(activity);
+
+            return await ProcessActivityAsync(activity, callback, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Primary adapter method for processing activities sent from channel.
+        /// </summary>
+        /// <param name="activity">The activity to process.</param>
+        /// <param name="callback">The BotCallBackHandler to call on completion.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>The response to the activity.</returns>
+        public async Task<InvokeResponse> ProcessActivityAsync(Activity activity, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            BotAssert.ActivityNotNull(activity);
+
+            _logger.LogInformation($"Received an incoming activity.  ActivityId: {activity.Id}");
+
+            using (var context = new TurnContext(this, activity))
+            {
+                await RunPipelineAsync(context, callback, cancellationToken).ConfigureAwait(false);
+
+                // Handle Invoke scenarios, which deviate from the request/response model in that
+                // the Bot will return a specific body and return code.
+                if (activity.Type == ActivityTypes.Invoke)
+                {
+                    var activityInvokeResponse = context.TurnState.Get<Activity>(InvokeReponseKey);
+                    if (activityInvokeResponse == null)
+                    {
+                        return new InvokeResponse { Status = (int)HttpStatusCode.NotImplemented };
+                    }
+                    else
+                    {
+                        return (InvokeResponse)activityInvokeResponse.Value;
+                    }
+                }
+
+                // For all non-invoke scenarios, the HTTP layers above don't have to mess
+                // withthe Body and return codes.
+                return null;
```

Comment:
>null [](start = 23, length = 4)

Can this be avoided?

Reply:
It matches the existing ProcessActivity behavior the bot is expecting. I can dig deeper into why, but for now I don't want to change a return that could break compatibility.